pkg/operator/controller/ingress/status.go

```diff
@@ -538,6 +544,45 @@ func formatConditions(conditions []*operatorv1.OperatorCondition) string {
 	return formatted
 }
 
+// computeIngressProgressingCondition computes the ingresscontroller's
+// "Progressing" status condition, which indicates if the ingresscontroller's
+// current state matches its desired state.
+func computeIngressProgressingCondition(conditions []operatorv1.OperatorCondition, ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
+	condition := operatorv1.OperatorCondition{
+		Type:   operatorv1.OperatorStatusTypeProgressing,
+		Status: operatorv1.ConditionFalse,
+	}
+	if ic.Status.EndpointPublishingStrategy.Type == operatorv1.LoadBalancerServiceStrategyType {
+		switch {
+		case ic.Status.EndpointPublishingStrategy.LoadBalancer == nil:
+			condition.Message = "status.endpointPublishingStrategy.loadBalancer is not set."
+			condition.Reason = "IncompleteStatus"
+			condition.Status = operatorv1.ConditionUnknown
+		case service == nil:
+			condition.Message = "LoadBalancer Service not created."
+			condition.Reason = "NoService"
+			condition.Status = operatorv1.ConditionTrue
+		default:
+			wantScope := ic.Status.EndpointPublishingStrategy.LoadBalancer.Scope
+			haveScope := operatorv1.ExternalLoadBalancer
+			if IsServiceInternal(service) {
+				haveScope = operatorv1.InternalLoadBalancer
+			}
+			if wantScope != haveScope {
+				message := fmt.Sprintf("Have load balancer with scope %q, want load balancer with scope %q.", haveScope, wantScope)
+				switch platform.Type {
+				case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
+					message = fmt.Sprintf("%s  You can delete the %s/%s service to proceed, and the service load-balancer will be deprovisioned and recreated.", message, service.Namespace, service.Name)
```

Comment:
I think as is, this message could be confusing to a cluster administrator:

1) I think `recreated` in this context sort of implies that the new load-balancer will have the same hostname and IP as the last load-balancer. Should we explicitly state that when deleting the load-balancer service, a new load-balancer will be stood up that will most likely have a new hostname and IP?

2) Should we make it clear to the cluster administrator that they are free to "undo" their scope change without any reprocussions? I think some more details in general could be useful here as to not confuse a cluster administrator.

Reply:
I've updated the message to convey these points.  Does it look all right?  (I hope it isn't _too_ verbose.)

Reply:
Looks fine to me :grin: